src/univariate/affine.jl

```diff
@@ -0,0 +1,332 @@
+```
+
+If `ρ` is a discrete univariate distribution, the probability mass function of the transformed 
+distribution is given by
+```math
+P(X = x) = P\\left(Z = \\frac{x-μ}{σ} \\right).
+```
+
+If `ρ` is a continuous univariate distribution with probability density function `f_Z`,
+the probability density function of the transformed distribution is given by
+```math
+f_X(x) = \\frac{1}{|σ|} f_Z\\left( \\frac{x-μ}{σ} \\right).
+```
+
+Generally, it is recommended to not use the `AffineDistribution` constructor but
+`+`, `-`, `*`, and `/` to construct distributions of affine transformations. The latter fall back
+to constructing an `AffineDistribution` but can return more optimized distributions, e.g.,
+if `ρ` is from a location-scale family.
+
+```julia
+d = σ * ρ + μ       # Create location-scale transformed distribution
+params(d)           # Get the parameters, i.e. (μ, σ, ρ)
+```
+"""
+struct AffineDistribution{N,S<:ValueSupport,Tμ,Tσ,D<:Distribution{ArrayLikeVariate{N},S}} <: Distribution{ArrayLikeVariate{N},S}
+    μ::Tμ
+    σ::Tσ
+    dist::D
+end
+
+
+#### Aliases
+@deprecate LocationScale(args...; kwargs...) AffineDistribution(args...)
+const ContinuousAffine{F,Tμ,Tσ,D} = AffineDistribution{F,Continuous,Tμ,Tσ,D}
+const DiscreteAffine{F,Tμ,Tσ,D} = AffineDistribution{F,Discrete,Tμ,Tσ,D}
+
+const UnivariateAffine{S,Tμ,Tσ,D} = AffineDistribution{Univariate,S,Tμ,Tσ,D}
+const MultivariateAffine{S,Tμ,Tσ,D} = AffineDistribution{Multivariate,S,Tμ,Tσ,D}
+
+
+#### Constructors
+
+"""
+    +(μ, dist)
+
+Return a version of `dist` that has been translated by `μ`.
+"""
+Base.:+(μ::Real, ρ::UnivariateDistribution) = AffineDistribution(μ, one(μ), ρ)
+
+function Base.:+(μ::AbstractArray{<:Real, N}, ρ::Distribution{ArrayLikeVariate{N}, S}) where {N, S}
+    if size(μ) ≠ size(dist) 
+        throw(DimensionMismatch("array and distribution have different sizes."))
+    end
+    return AffineDistribution(μ, Diagonal(Ones(μ)), ρ)
+end
+
+
+"""
+    *(σ, dist)
+
+Return a version of `dist` that has been scaled by `σ`, where `σ` can be any real number or 
+matrix.
+"""
+function Base.:*(σ::Real, ρ::UnivariateDistribution)
+    if iszero(σ)
+        throw(ArgumentError("scale must be non-zero"))
+    end
+    return AffineDistribution(zero(σ), σ, ρ)
+end
+
+function Base.:*(σ::Real, ρ::Distribution)
+    if iszero(σ)
+        throw(ArgumentError("scale must be non-zero"))
+    end
+    return AffineDistribution(Zeros{eltype(ρ)}(size(ρ)), σ, ρ)
+end
+
+function Base.:*(σ::AbstractMatrix{<:Real}, ρ::MultivariateDistribution)
+    if iszero(σ)
+        throw(ArgumentError("scale must be non-zero"))
+    elseif size(σ)[end] ≠ size(ρ)[1]
+        throw(DimensionMismatch(
+            "scaling matrix and distribution have incompatible dimensions" *
+            "$(size(σ)) and $(size(ρ)))"
+        ))
+    end
+    return AffineDistribution(Zeros{eltype(ρ)}(σ), σ, ρ)
+end
+Base.:*(σ::AbstractVector{<:Real}, ρ::MultivariateDistribution) = Diagonal(σ) * ρ
```

Comment:
I think this should not be defined since `σ * Z` is not defined either where the law of `Z` is `ρ`.
```suggestion
```

Reply:
Isn't this a default method for e.g. Normal distributions, though?

Actually, maybe it would make sense to define `σ * ρ` as `σ .* ρ` when the dimensions line up? So we can allow scaling the marginals for arbitrary distributions.

Reply:
> Isn't this a default method for e.g. Normal distributions, though?

No, there's no such method, there is only https://github.com/JuliaStats/Distributions.jl/blob/e94d6704386d1f753b28011067af03990303b14c/src/multivariate/mvnormal.jl#L295. Precisely for the reason that I mentioned in my comment: `rand(5) * rand(5)` is not defined but `rand(2, 5) * rand(5)` is.

> Actually, maybe it would make sense to define σ * ρ as σ .* ρ when the dimensions line up? 

There's a hard no from my side. In my experience implicit broadcasting usually is unintuitive, surprising for users, and difficult to handle for developers.

`Diagonal(rand(5)) * rand(5)` is clear and simple. As another example, we deprecated `MvNormal(::AbstractVector)` since the implicit convention `MvNormal(x::AbstractVector) = MvNormal(zeros(length(x)), Diagonal(x .^ 2))` is not clear and caused a lot of confusion among users - the vector could also denote eg. the mean. However, `MvNormal(Diagonal(rand(5)))` is less surprising - it returns a normal distribution with zero mean and the given diagonal covariance matrix. Similarly, I think it is clearer to provide (diagonal) matrices here instead of performing any implicit broadcasting.

Reply:
Makes sense. Removed this